src/crimson/osd/ops_executer.h

```diff
@@ -263,28 +267,41 @@ auto OpsExecuter::with_effect_on_obc(
 }
 
 template <typename MutFunc>
-OpsExecuter::osd_op_ierrorator::future<> OpsExecuter::flush_changes(
+OpsExecuter::rep_op_fut_t OpsExecuter::flush_changes(
   MutFunc&& mut_func) &&
 {
   const bool want_mutate = !txn.empty();
   // osd_op_params are instantiated by every wr-like operation.
   assert(osd_op_params || !want_mutate);
   assert(obc);
-  auto maybe_mutated = interruptor::make_interruptible(osd_op_errorator::now());
+  rep_op_fut_t maybe_mutated =
+    interruptor::make_ready_future<rep_op_fut_tuple>(
+	std::make_tuple(seastar::now(),
+			interruptor::make_interruptible(osd_op_errorator::now())));
   if (want_mutate) {
-    maybe_mutated = std::forward<MutFunc>(mut_func)(std::move(txn),
+    auto [submitted, all_completed] = std::forward<MutFunc>(mut_func)(std::move(txn),
                                                     std::move(obc),
                                                     std::move(*osd_op_params),
                                                     user_modify);
+    maybe_mutated = interruptor::make_ready_future<rep_op_fut_tuple>(
+	std::make_tuple(
```

Comment:
with the proposed parameter pack change, we can dispense with the `make_tuple()` call.